### PR TITLE
Proposal: add replace directive to go.mod to use local version of v3/* modules

### PR DESCRIPTION
**Pull request by VAveryanov8**
This simplifies workflow when it's needed to make some changes in v3/* modules.

Relates to #4109 

---

Please make sure that:
- Code is split to commits that address a single change
- Commit messages are informative
- Commit titles have module prefix
- Commit titles have issue nr. suffix

### CODE REVIEW

**Comment by VAveryanov8** [comment]
Ok, here is how it works: 
1. From the SM perspective, when scylla-manager repository is pulled, then `scylla-manager/go.mod` file is the main module and all go tooling will consider `replace` directives of `scylla-manger/go.mod`. 
2. When some external module (e.g. siren) is using `v3/pkg/managerclient`, then main module is `siren/go.mod` and `replace` directives of `scylla-manager/go.mod` and `v3/pkg/managerclient/go.mod` are ignored, but instead go tooling uses versions from their `require` sections. 

Here is short example to show this behavior. Imaging we have some `managerclient_test` module, looking like this: 
**main.go:**
```go
package main

import (
	"github.com/scylladb/scylla-manager/v3/pkg/managerclient"
)

func main() {
	_, _ = managerclient.NewClient("example.com")
}

```
We pulled managerclient package from this branch (by commit hash): 
```bash 
go get github.com/scylladb/scylla-manager/v3/pkg/managerclient@0cfb5ab6da3bdc631afc138cbf4c6ba2d7cba24a 
``` 

Then go.mod will look like this (i've stripped all the deps except scylla-manager/v3*): 
```go
module managerclienttest

go 1.23.2

require github.com/scylladb/scylla-manager/v3/pkg/managerclient v0.0.0-20241118090522-0cfb5ab6da3b

require (
	....
	github.com/scylladb/scylla-manager/v3/pkg/util v0.0.0-20241030073626-e409ae491c83 // indirect
	github.com/scylladb/scylla-manager/v3/swagger v0.0.0-20241030073626-e409ae491c83 // indirect
	...
)

```

We can see that managerclient version is from this branch, but `v3/pkg/util` and `v3/swagger` have version that are listed in `require` section of `managerclient/go.mod` from this branch. 

This means, that if we want to update managerclient dependencies for the external usage, then we need to explicitly update `managerclient/go.mod`, but when working on `scylla-manger` only local versions of `v3/*` modules will be used and there is no need to do several pull requests when we need to update `v3/swagger` for example.  

P.S.
This behavior is described in [go.mod reference](https://go.dev/ref/mod#go-mod-file-replace)

> replace directives only apply in the main module’s go.mod file and are ignored in other modules. See [Minimal version selection](https://go.dev/ref/mod#minimal-version-selection) for details.

**Comment by VAveryanov8** [comment]
> So if I understand correctly, we should use replace directives only in the scylla manager module go.mod file, but not in the managerclient module go.mod file?

`replace` directive only applies to main module, so if (in theory) we want to use `managerclient` as a main module, e.g adding `main` package  in `managerclient`, then there is a place for `replace` directive, otherwise indeed it's not needed.

> Also, this could still create problems when someone would want to import scylla manager module.  

You mean problems like build version of scylla-manager is different from was what imported as module, because of the different dependencies version in go.mod?  Yes, additional effort to update go.mod file is still  be required in any case I think :( 

So in a nutshell by adding `replace` directive we can simplify the way we work with scylla manager, when making changes in both `v3/*` modules and main scylla-manager module. But still the additional effort to keep go.mod file updated is required.

BTW: do we keep managerclient/go.mod up to date with latest versions of `v3/swagger` and `v3/pkg/util`?

**Comment by karol-kokoszka** [comment]
Maybe go.work is not a bad option at all (even if it's commit to the repo).
